sqlglot/dialects/hive.py

```diff
@@ -451,6 +471,37 @@ def _parse_types(
 
             return this
 
+        def _parse_alter_table_change(self) -> t.Optional[exp.Expression]:
+            self._match(TokenType.COLUMN)
+            this = self._parse_column()
```

Comment:
I'm not sure if this will produce a correct AST. This results in a `Column` node, but these are really **identifiers**. For instance:

```python
>>> import sqlglot
>>> sqlglot.parse_one("ALTER TABLE integers ALTER COLUMN i SET DATA TYPE VARCHAR")
Alter(
  this=Table(
    this=Identifier(this=integers, quoted=False)),
  kind=TABLE,
  actions=[
    AlterColumn(
      this=Identifier(this=i, quoted=False),
      dtype=DataType(this=Type.VARCHAR, nested=False))])
```

The distinction is made because they're generally treated differently in the optimizer modules.